### PR TITLE
Tooling & PHP Version Updates

### PR DESCRIPTION
**Pull request by sixlive**
## Description

Overall this PR is prepping for release 2.0 with the following changes:

- Drop support for PHP 7.x, 8.0, 8.1 (unsupported versions)
- Add Github Actions for tests and formatting
- Migrate to [Pest](https://pestphp.com) for testing
  - Migrate tests to Pest format
- Drop support for TravisCI
- Setup for [Rector](https://getrector.com) and [Pint](https://laravel.com/docs/11.x/pint)
  - Apply Rector and Pint formatting
 - Add git hooks for DX

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,22 +1,13 @@
...
-        <testsuite name="Geocodio Test Suite">
-            <directory>tests</directory>
-        </testsuite>
-    </testsuites>
-    <filter>
-        <whitelist>
-            <directory suffix=".php">src/</directory>
-        </whitelist>
-    </filter>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bootstrap="vendor/autoload.php" backupGlobals="false" colors="true" processIsolation="false" stopOnFailure="false" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/11.3/phpunit.xsd" cacheDirectory=".phpunit.cache" backupStaticProperties="false">
```
Comment (MiniCodeMonkey) [inline phpunit.xml.dist:2]: @sixlive Can this file be removed now, since we're using Pest?

File: src/Geocodio.php [hunk 2]
```diff
@@ -149,26 +150,28 @@ private function performSingleRequest(string $url, $query, array $queryParameter
...
+    private function handleException(\Throwable $e): void
+    {
         $response = $e instanceof RequestException && $e->hasResponse() ? $e->getResponse() : null;
 
         $errorMessage = 'Error';
         $errorCode = 0;
 
-        if ($response) {
-            $json = @json_decode((string)$response->getBody());
+        if ($response instanceof \Psr\Http\Message\ResponseInterface) {
```
Comment (MiniCodeMonkey) [inline src/Geocodio.php:173]: 👍

File: src/Geocodio.php [hunk 3]
```diff
@@ -180,13 +183,15 @@ private function handleException(Exception $e) {
...
+    {
+        return json_decode((string) $response->getBody());
     }
 
-    private function getHeaders(): array {
+    private function getHeaders(): array
+    {
         return [
-            'User-Agent' => 'geocodio-library-php/1.2.0'
+            'User-Agent' => 'geocodio-library-php/1.2.0',
```
Comment (MiniCodeMonkey) [inline src/Geocodio.php:194]: I guess this is fine since it is addressed in #11 instead

File: tests/ArchTest.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+namespace Geocodio\Tests;
+
+describe('Arch presets', function (): void {
+    arch('security preset')->preset()->security();
+    arch('php preset')->preset()->php();
+});
+
+describe('Custom relaxed preset', function (): void {
+    arch('No final classes')
+        ->expect('EchoLabs\Sparkle')
```
Comment (MiniCodeMonkey) [inline tests/ArchTest.php]: Should this namespace be updated to `Geocodio`?

File: whisky.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+    "disabled": [],
+    "hooks": {
+        "pre-commit": [
```
Comment (MiniCodeMonkey) [inline whisky.json:4]: Neat!! 👍

**Review by MiniCodeMonkey — APPROVED** [review]

File: phpunit.xml.dist (hunk 1 above)
Comment (sixlive) [reply phpunit.xml.dist:2]: Pest uses PHPUnit under the hood so the config file is still used

File: src/Geocodio.php (hunk 3 above)
Comment (sixlive) [reply src/Geocodio.php:194]: Yeah, we address this in #11

File: tests/ArchTest.php (hunk 4 above)
Comment (sixlive) [reply tests/ArchTest.php]: Good catch, copypasta

**Pull request merged**